File: rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int b) {
        int k = b % nums.length;
        reverse(nums,0,nums.length-1);
        reverse(nums,0,k-1);
        reverse(nums,k,nums.length-1);
    }
    public void reverse (int [] nums , int s, int e ){
        
        while (e>s){
          int  temp = nums[s];
            nums[s] = nums[e];
            nums[e] = temp;
            s++;
            e--;
        }
    }
}

File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        def reverse(array, start, end):
            while start<=end:
                array[start], array[end] = array[end], array[start]
                start+=1
                end-=1
            return array
        k = k%len(nums)
        
        nums.reverse()
        reverse(nums,0,k-1)
        reverse(nums,k, len(nums)-1)
        return nums


        


        